core/settings.lua

```diff
@@ -28,6 +28,7 @@ function settings:_init ()
             ]]):format(language, language, language, language))
          end
          fluent:set_locale(language)
+         SU.setenv("LANG", language)
```

Comment:
While this could work for some language, the correct approach might be trickier:
- `LANG`expect a Unix locale (eg. `de_DE`)
- Our current `document.language` is only a two letter code (eg. `de`), which may be under-resolved
- If we move our `document.language` at some point to possibly fit either BCP47 or ICU locales, it still likely wouldn't be a valid Unix locale (e.g. consider e.g. BCP47 code `sr-Latn-RS`, ICU locale `sr_Latn_RS`, Unix locale `sr_RS@Latn`)

So we might have to consider conversion utilities (BCP47 -> ICU local -> Unix locale)

Reply:
While there are one-to-one canonicalization rules between BCP47 languages and ICU locales (which our justenoughicu wrapper even implements), I can't find any clear source for mapping to Unix/Posix locales (setlocale) as expected by LANG. So it might be a a hornest's nest.

(BTW, Why would we want it? For Lua `os.date` etc. to be localized? I'm not sure there's a good use case for it)

Reply:
Yes, this whole maneuver is dependent on us starting to use proper locales (BCP47 or whatever) internally and on adding a mapping to the `setenv()` function that converts whatever than scheme is to a Unix locale.

I'm already using this is production with a small hack to map the languages I use. I have documents where one class is used to render inputs in multiple languages, and for example the inprint page has date information. This is rendered with the system `date` command, but in only outputs the correct localized string if the locale is set. This already works if I supply the right mapping. I also have other tools besides `date` that output content that gets ingested during the SILE render process (for figures, music, etc.) and is sensitive to the locale.

The real question is whether there is ever a case that setting the LANG will produce undesired results. For it to work the system tools need to have support for the expected locales compiled using `locale-gen` or similar, but would the user ever *not* expect it to work this way? I couldn't think of any time this would be objectively wrong.

Reply:
Yes, except that instead of having our own custom "just enough" wrapper for ICU we're going to have access to the full ICU API surface via Rust crates (and of course can expose whatever bits of it we want on the Lua side, which is basically what we have now in `justenough...` except we won't need the C wrapper.

Reply:
> While this could work for some language

Still true, but I'm actually going to merge this before #1641 understanding that it's not going to work for all locales yet, but there is no point I see in waiting on proper local parsing and it might even make it easier to test locale parsing to have this in place.